File: Data Structures & Algorithms/top-k-elements-in-list/submission-3.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        mp = {}

        for n in nums:
            mp[n] = 1+mp.get(n, 0);
        
        minHeap = []
        for num in mp.keys():
            heapq.heappush(minHeap, (mp[num], num))
            if len(minHeap) > k:
                heapq.heappop(minHeap)

        res = []
        for i in range(k):
            res.append(heapq.heappop(minHeap)[1])
        return res

File: Data Structures & Algorithms/maximum-subarray/submission-3.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int maxSum = 0;
        int res = nums[0];
        for(int n:nums) {
            maxSum = n + max(0, maxSum);
            res = max(res, maxSum);
        }

        return res;
    }
};